PlayCover/Utils/NavigationStack.swift

```diff
@@ -0,0 +1,114 @@
+//
+//  NavigationStack.swift
+//  PlayCover
+//
+//  Created by Amir Mohammadi on 11/2/1401 AP.
+//
+
+import SwiftUI
+
+enum StackNavigationTransition: Equatable {
+    static func == (lhs: StackNavigationTransition, rhs: StackNavigationTransition) -> Bool {
+        lhs.id == rhs.id
+    }
+
+    case none, defaultTranisition, custom(AnyTransition)
+
+    var id: String {
+        switch self {
+        case .none: return "none"
+        case .defaultTranisition: return "default"
+        case .custom: return "custom"
+        }
+    }
+
+    var anyTransition: AnyTransition {
+        switch self {
+        case .none:
+            return AnyTransition.identity
+        case .defaultTranisition:
+            /// In this case you need to wrap `showingSubview = true` in the
+            /// root of the view view in a `withAnimation(.spring()) { ... }`
+            return AnyTransition.move(edge: .trailing)
+        case .custom(let transition):
+            return transition
+        }
+    }
+}
+
+struct StackNavigationView<RootContent>: View where RootContent: View {
+    let rootView: () -> RootContent
+
+    @Binding var currentSubview: AnyView
+    @Binding var showingSubview: Bool
+    @State var transition: StackNavigationTransition
+
+    init(currentSubview: Binding<AnyView>,
+         showingSubview: Binding<Bool>,
+         transition: StackNavigationTransition,
+         @ViewBuilder rootView: @escaping () -> RootContent) {
+        self._currentSubview = currentSubview
+        self._showingSubview = showingSubview
+        self.transition = transition
+        self.rootView = rootView
+    }
+
+    var body: some View {
+        VStack {
+            if !showingSubview {
+                rootView()
+                    .zIndex(-1)
+                    .transition(transition == .defaultTranisition
+                                ? .move(edge: .leading)
+                                : transition.anyTransition)
+            } else {
+                StackNavigationSubview(currentSubview: $currentSubview,
+                                       isVisible: $showingSubview,
+                                       transition: transition) {
+                    currentSubview
+                        .transition(transition.anyTransition)
+                        .zIndex(1)
+                }
```

Comment:
Is there any reason declaring and using StackNavigationSubview?

Reply:
Actually I don't remember why I did it but now I see it's redundant as you said. Thanks for mentioning it!